packages/context/src/resolver.ts

```diff
@@ -100,7 +101,8 @@ function resolve<T>(
         return injection.resolve(ctx, injection, s);
       } else {
         // Default to resolve the value from the context by binding key
-        return ctx.getValueOrPromise(injection.bindingKey, {
+        const key = injection.bindingSelector as BindingAddress;
```

Comment:
What if bindingSelector is a filter? It is still not clear to me how we can be sure that it will be always a single address and binding filters will never reach this code. Can you explain please?

Reply:
_https://github.com/strongloop/loopback-next/pull/2122/files#r248976288_

What if bindingSelector is a filter? It is still not clear to me how we can be sure that it will be always a single address and binding filters will never reach this code. Can you explain please?

Reply:
`bindingSelector` is always an `BindingAddress` for the default `resolve` function. Those with a binding filter are handled by specific `resolve*`.

Reply:
Can you add an `assert` to verify that assumption at runtime please?

```ts
const selector = injection.bindingSelector;
if (!isBindingAddress(selector)) {
  throw new AssertionError(...);
}
const key = selector; // no cast is needed
```

Reply:
Would it perhaps make more sense to simply move the code from `resolveValuesByFilter` directly into `resolve`, so that `@inject(filter)` does not need any custom resolver?